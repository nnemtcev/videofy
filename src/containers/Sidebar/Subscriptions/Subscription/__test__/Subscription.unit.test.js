import { shallow } from "enzyme";
import React from "react";

import { Subscription } from "../Subscription.js";

describe("Subscription", () => {
  test("empty Subscription component renders", () => {
    const wrapper = shallow(<Subscription />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Subscription component with label renders", () => {
    const wrapper = shallow(<Subscription label="Coursera" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Subscription component with broadcasting renders", () => {
    const wrapper = shallow(<Subscription label="Coursera" broadcasting />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Subscription component with label, non-broadcasting but with new videos", () => {
    const wrapper = shallow(
      <Subscription label="Coursera" amountNewVideos={7} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
