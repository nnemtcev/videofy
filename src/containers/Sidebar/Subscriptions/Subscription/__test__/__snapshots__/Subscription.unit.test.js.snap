// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Subscription component with broadcasting renders 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Coursera
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription Subscription component with label renders 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Coursera
      </span>
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription Subscription component with label, non-broadcasting but with new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Coursera
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      7
    </span>
  </div>
</MenuItem>
`;

exports[`Subscription empty Subscription component renders 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;
